kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/KroxyliciousTesterBuilder.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.test.tester;
+
+import java.util.function.Function;
+
+import io.kroxylicious.proxy.config.Configuration;
+import io.kroxylicious.proxy.config.ConfigurationBuilder;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Kroxylicious strives to offer high quality testing tools which allow filter authors to focus on their business logic.
+ * However, as Kroxylicious is very configurable the so is the Tester. Most of the time users do not care about a lot of the configuration options
+ * and thus a builder, which can provide sensible defaults, provides the simplest API to configure things.
+ */
+public class KroxyliciousTesterBuilder {
+
+    private String trustStoreLocation = null;
+    @Nullable
+    private String trustStorePassword = null;
+    private Function<Configuration, AutoCloseable> kroxyliciousFactory = DefaultKroxyliciousTester::spawnProxy;
+    private DefaultKroxyliciousTester.ClientFactory clientFactory = (clusterName, defaultConfiguration) -> new KroxyliciousClients(defaultConfiguration);
+    private ConfigurationBuilder configurationBuilder;
+
+    /**
+     * Supplies the builder used to configure the kroxylicious instance to be used in tests.
+     * @param configurationBuilder the builder instance ussd to configure the kroxylicious instance.
+     * @return the current instance of the builder for chaining in a fluent fashion.
+     */
+    public KroxyliciousTesterBuilder setConfigurationBuilder(ConfigurationBuilder configurationBuilder) {
+        this.configurationBuilder = configurationBuilder;
```

Comment:
Its not legal for configurationBuilder to be null.  Would it make sense for the builder to enforce that it is set?